src/sys/unix/waker.rs

```diff
@@ -1,5 +1,5 @@
 #[cfg(all(
-    not(mio_unsupported_force_poll_poll),
+    not(any(target_os = "solaris", mio_unsupported_force_poll_poll)),
```

Comment:
Not needed.

Reply:
I got following error without this change:
````
error: field `waker` is never read
  --> src/sys/unix/waker.rs:38:9
   |
37 |     pub struct Waker {
   |                ----- field in this struct
38 |         waker: WakerInternal,
   |         ^^^^^
   |
   = note: `Waker` has a derived impl for the trait `Debug`, but this is intentionally ignored during dead code analysis
note: the lint level is defined here
  --> src/lib.rs:6:5
   |
6  |     dead_code
   |     ^^^^^^^^^

error: associated items `new` and `wake` are never used
  --> src/sys/unix/waker.rs:42:16
   |
41 |     impl Waker {
   |     ---------- associated items in this implementation
42 |         pub fn new(selector: &Selector, token: Token) -> io::Result<Waker> {
   |                ^^^
...
48 |         pub fn wake(&self) -> io::Result<()> {
   |                ^^^^
```

Reply:
I think it uses `pub use self::poll::Waker;` and thus cannot use `pub use self::fdbased::Waker`.

Reply:
@Thomasdezeeuw How can possibly following attribute be `false` on Solaris?
```
#[cfg(all(
    not(mio_unsupported_force_poll_poll),
    not(all(
        not(mio_unsupported_force_waker_pipe),
        any(
            target_os = "freebsd",
            target_os = "ios",
            target_os = "macos",
            target_os = "tvos",
            target_os = "watchos",
        )
    ))
))]
```